manifests/dev/vsphere-7.0u2/guestcluster/1.17/pvcsi.yaml

```diff
@@ -49,14 +55,20 @@ metadata:
   name: vsphere-csi-node-role
   namespace: {{ .PVCSINamespace }}
 rules:
-  - apiGroups:
-    - "policy"
-    resources:
-    - podsecuritypolicies
-    verbs:
-    - use
-    resourceNames:
-    - vmware-system-privileged
+  - apiGroups: ["policy"]
+    resources: ["podsecuritypolicies"]
+    verbs: ["use"]
+    resourceNames: ["vmware-system-privileged"]
+---
+kind: ClusterRole
```

Comment:
`k8sOrchestrator` init watches on configmaps https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go#L192 which internally calls https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/kubernetes/informers.go#L93 which is not namespace specific. Therefore, `ClusterRole` is required to watch on configmaps.

Reply:
ConfigMap is a namespace scoped object. I did not quite get why just Role is not good enough.

Reply:
Currently our informer implementation in https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/kubernetes/informers.go#L93 does not take in a namespace to listen on. It listens to configmaps in all namespaces. If I use a `Role`, the code errors out at this point stating that it doesn't have permissions to listen on all namespaces. Our admission controller also uses a `ClusterRole` to read the FSS values.

Reply:
Its a bad idea to listen to all config map changes not related to CSI. We should fix that.

Reply:
Then we need to fix it before 2.1 release as admission controller uses the exact same scenario.
cc: @divyenpatel

Reply:
@shalini-b we have the capability to create config-map informer for a specific namespace. We should create a configmap informer using this method.

```
// NewConfigMapInformer constructs a new informer for ConfigMap type.
// Always prefer using an informer factory to get a shared informer instead of getting an independent
// one. This reduces memory footprint and number of connections to the server.
func NewConfigMapInformer(client kubernetes.Interface, namespace string, resyncPeriod time.Duration, indexers cache.Indexers) cache.SharedIndexInformer {
	return NewFilteredConfigMapInformer(client, namespace, resyncPeriod, indexers, nil)
}
```

Reply:
Tracking this as a separate bug. Resolving this issue for now.

Reply:
Can you file an issue and also add a TODO here to fix the issue?

Reply:
Done.